### DISCUSSION TITLE
Debugging - How To

### DISCUSSION THREAD

**Discussion by dcreedon** [OpenTTD/OpenTTD#12453] [Q&A]
Hi,
 I'm running Debian and have successfully downloaded, compiled and run OpenTTD. All good so far. I'm using VSCode what is the best way to debug OpenTTD. For example, I want to set a breakpoint here in the first statement of openttd_main,  I can't do run start debugging as I either don't have it setup correctly or need to try another way...  any help appreciated.

![image](https://github.com/OpenTTD/OpenTTD/assets/6379032/7844f3c7-10c4-4d70-ab97-e5cad408d239)

**Comment by zacc** [comment]
I would like to know too; but this seems like a good task for a YouTuber/blogger.
